pyodk/_endpoints/submission_attachments.py

```diff
@@ -0,0 +1,168 @@
+        self.session: Session = session
+        self.default_project_id: int | None = default_project_id
+        self.default_form_id: str | None = default_form_id
+        self.default_instance_id: str | None = default_instance_id
+
+    def list(
+        self,
+        instance_id: str | None = None,
+        form_id: str | None = None,
+        project_id: int | None = None,
+    ) -> list[SubmissionAttachment]:
+        """
+        Show all required submission attachments and their upload status.
+
+        :param instance_id: The instanceId of the submission being referenced.
+        :param form_id: The xmlFormId of the Form being referenced.
+        :param project_id: The id of the project the Submissions belong to.
+
+        :return: A list of the object representation of all Submission
+            attachment metadata.
+        """
+        try:
+            pid = pv.validate_project_id(project_id, self.default_project_id)
+            fid = pv.validate_form_id(form_id, self.default_form_id)
+            iid = pv.validate_form_id(instance_id, self.default_instance_id)
+        except PyODKError as err:
+            log.error(err, exc_info=True)
+            raise
+
+        response = self.session.response_or_error(
+            method="GET",
+            url=self.session.urlformat(
+                self.urls.list, project_id=pid, form_id=fid, instance_id=iid
+            ),
+            logger=log,
+        )
+        data = response.json()
+        return [SubmissionAttachment(**r) for r in data]
+
+    def get(
+        self,
+        file_name: str,
+        instance_id: str,
+        form_id: str | None = None,
+        project_id: int | None = None,
+    ) -> bytes:
+        """
+        Read Submission metadata.
+
+        :param file_name: The file name of the Submission attachment being referenced.
+        :param instance_id: The instanceId of the Submission being referenced.
+        :param form_id: The xmlFormId of the Form being referenced.
+        :param project_id: The id of the project this form belongs to.
+
+        :return: The attachment bytes for download.
+        """
+        try:
+            pid = pv.validate_project_id(project_id, self.default_project_id)
+            fid = pv.validate_form_id(form_id, self.default_form_id)
+            iid = pv.validate_instance_id(instance_id, self.default_instance_id)
+        except PyODKError as err:
+            log.error(err, exc_info=True)
+            raise
+
+        response = self.session.response_or_error(
+            method="GET",
+            url=self.session.urlformat(
+                self.urls.get,
+                project_id=pid,
+                form_id=fid,
+                instance_id=iid,
+                fname=file_name,
+            ),
+            logger=log,
+        )
+        return response.content
+
+    def upload(
+        self,
+        file_path_or_bytes: PathLike | str | bytes,
```

Comment:
Please follow the `forms.create` process where it's assumed file paths are passed in. If using bytes is needed please open a ticket for that so we can think about doing that consistently across endpoints that currently accept file paths.